File: 645-set-mismatch/set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int [] freq = new int[nums.length + 1];
        int duplicate = 0;
        int missing = 0;
        for (int i = 0; i < nums.length; i++){
            freq[nums[i]]++;
        }

        for(int i = 1; i<freq.length; i++){
            if(freq[i] == 2){
                duplicate = i;
            }
            else if(freq[i] == 0){
                missing = i;
            }
        }
        return new int[] {duplicate , missing};
    }
}

File: 645-set-mismatch/set-mismatch.py
class Solution:
    #Rehan I Baig gitHub: @raaayhaan
    def findErrorNums(self, nums: List[int]) -> List[int]:
        duplicate = 0
        missing = 0
        freq = [0] * (len(nums) + 1)
        for num in nums:
            freq[num] += 1
        for i in range (1, len(freq)):
            if freq[i] == 2 :
                duplicate = i
            elif freq[i] == 0 :
                missing = i
        return [duplicate , missing]